cloudinit/config/schema.py

```diff
@@ -410,34 +420,53 @@ def _get_property_doc(schema: dict, prefix="    ") -> str:
     """Return restructured text describing the supported schema properties."""
     new_prefix = prefix + '    '
     properties = []
-    for prop_key, prop_config in schema.get('properties', {}).items():
-        # Define prop_name and description for SCHEMA_PROPERTY_TMPL
-        description = prop_config.get('description', '')
-
-        # Define prop_name and description for SCHEMA_PROPERTY_TMPL
-        properties.append(
-            SCHEMA_PROPERTY_TMPL.format(
-                prefix=prefix,
-                prop_name=prop_key,
-                description=_parse_description(description, prefix),
-                prop_type=_get_property_type(prop_config),
+    property_keys = [
+        schema.get("properties", {}),
+        schema.get("patternProperties", {}),
+    ]
+
+    for props in property_keys:
+        for prop_key, prop_config in props.items():
+            # Define prop_name and description for SCHEMA_PROPERTY_TMPL
```

Comment:
I don't think so. Properties definitions look something like:
```
'properties': {
    'thing1': {
        'type': 'object',
        ...
    },
    'thing2': {
        ....
    }
}
```
With this example, first time through, `prop_key` would be `thing1` and `prop_config` would be the dictionary it defines. The only time we wouldn't get a truthy value for prop_conig is if we had a definition of one of the properties like `"thing1": {}` or `"thing1": None`, which is weird and doesn't adhere to the metaschema.

If we want to do validation within the function, we could bail if prop_config isn't an instance of dict/mapping, but what to do then? I don't think we should just return given that something is wrong here. We could throw a ValueError rather than the AttributeError we'll naturally get, but I think I prefer leaving the responsibility of anything invalid to the caller. This is why we have the warning on [line 529](https://github.com/canonical/cloud-init/pull/1144/files#diff-3dab74160d0d2e18837e4c271eb03209be4263cf51d44cb8ca78bb739fd3f68cR529)

Reply:
Oops misread the looping. I thought prop_key and prop_config were actually going to be values like:
prop_key = "properties"
prop_cfg = {} but you are already going through items() of the properties under the "properties" key here.
+1 disregard.